CoreScriptsRoot/CoreScripts/PurchasePromptScript2.lua

```diff
@@ -269,136 +303,268 @@ local function createImageButtonWithText(name, position, image, imageDown, text,
+			Gui.CancelButton.ScaleType = scaleType
+			Gui.CancelButton.SliceCenter = sliceCenter
+			Gui.CancelButton.Visible = false
+			Gui.CancelButton.Parent = Gui.ContainerFrame
+
+			Gui.BuyRobuxButton = createImageButtonWithText3D("BuyRobuxButton", UDim2.new(0, 10, 0, 225), VR_BUTTON, VR_BUTTON_DOWN, "Buy", Enum.Font.SourceSansBold, Enum.FontSize.Size36)
+			Gui.BuyRobuxButton.ScaleType = scaleType
+			Gui.BuyRobuxButton.SliceCenter = sliceCenter
+			Gui.BuyRobuxButton.Visible = false
+			Gui.BuyRobuxButton.Parent = Gui.ContainerFrame
+
+			Gui.BuyBCButton = createImageButtonWithText3D("BuyBCButton", UDim2.new(0, 10, 0, 225), VR_BUTTON, VR_BUTTON_DOWN, "Upgrade", Enum.Font.SourceSansBold, Enum.FontSize.Size36)
+			Gui.BuyBCButton.ScaleType = scaleType
+			Gui.BuyBCButton.SliceCenter = sliceCenter
+			Gui.BuyBCButton.Visible = false
+			Gui.BuyBCButton.Parent = Gui.ContainerFrame
+
+			Gui.FreeButton = createImageButtonWithText3D("FreeButton", UDim2.new(0, 10, 0, 225), VR_BUTTON, VR_BUTTON_DOWN, "Take Free", Enum.Font.SourceSansBold, Enum.FontSize.Size36)
+			Gui.FreeButton.ScaleType = scaleType
+			Gui.FreeButton.SliceCenter = sliceCenter
+			Gui.FreeButton.Visible = false
+			Gui.FreeButton.Parent = Gui.ContainerFrame
+
+			Gui.OkButton = createImageButtonWithText3D("OkButton", UDim2.new(0.25, 0, 0, 225), VR_BUTTON, VR_BUTTON_DOWN, "Ok", Enum.Font.SourceSansBold, Enum.FontSize.Size36)
+			Gui.OkButton.ScaleType = scaleType
+			Gui.OkButton.SliceCenter = sliceCenter
+			Gui.OkButton.Size = UDim2.new(0.5, 0, 0, Gui.OkButton.Size.Y.Offset)
+			Gui.OkButton.Visible = false
+			Gui.OkButton.Parent = Gui.ContainerFrame
+
+			Gui.OkPurchasedButton = createImageButtonWithText3D("OkPurchasedButton", BTN_M_POS_VR, VR_BUTTON, VR_BUTTON_DOWN, "Ok", Enum.Font.SourceSansBold, Enum.FontSize.Size36)
+			Gui.OkPurchasedButton.ScaleType = scaleType
+			Gui.OkPurchasedButton.SliceCenter = sliceCenter
+			Gui.OkPurchasedButton.Size = UDim2.new(0.5, 0, 0, Gui.OkPurchasedButton.Size.Y.Offset)
+			Gui.OkPurchasedButton.Visible = false
+			Gui.OkPurchasedButton.Parent = Gui.ContainerFrame
+
+		Gui.PurchaseFrame = createFrame("PurchaseFrame", UDim2.new(1, 0, 1, 0), nil, 1, nil)
+		Gui.PurchaseFrame.Visible = false
+		Gui.PurchaseFrame.Parent = Gui.PurchaseDialog
+
+			Gui.PurchaseText = createTextLabel("PurchaseText", nil, UDim2.new(0.5, 0, 0.5, -100), Enum.Font.SourceSans, Enum.FontSize.Size60, "Purchasing")
+			Gui.PurchaseText.Parent = Gui.PurchaseFrame
+
+			Gui.LoadingFrames = {}
+			local scaleFactor = 3
+			local xOffset = -40 * scaleFactor
+			for i = 1, 3 do
+				local frame = createFrame3D("Loading", UDim2.new(0, 16 * scaleFactor, 0, 16 * scaleFactor), UDim2.new(0.5, xOffset, 0.5, -50), 0, Color3.new(132/255, 132/255, 132/255))
+				table.insert(Gui.LoadingFrames, frame)
+				frame.Parent = Gui.PurchaseFrame
+				xOffset = xOffset + (32 * scaleFactor)
+			end
+
+	return Gui
+end
```

Comment:
Hm, we could. I wanted to keep them separate in case the 2D design changes. To account for all of the potential cases where the 2D version changes would be about the same amount of code...